File: problems/1656-design-an-ordered-stream/solution.java
import java.util.ArrayList;
import java.util.List;

class OrderedStream {

    private String[] stream;
    private int ptr;
    private int n;

    public OrderedStream(int n) {
        this.stream = new String[n + 1];
        this.ptr = 1;
        this.n = n;
    }

    public List<String> insert(int idKey, String value) {
        stream[idKey] = value;
        List<String> result = new ArrayList<>();
        while (ptr <= n && stream[ptr] != null) {
            result.add(stream[ptr]);
            ptr++;
        }
        return result;
    }
}

File: problems/1656-design-an-ordered-stream/solution.py
class OrderedStream:

    def __init__(self, n: int):
        self.stream = [None] * (n + 1)
        self.ptr = 1

    def insert(self, idKey: int, value: str) -> list[str]:
        self.stream[idKey] = value
        result = []
        while self.ptr < len(self.stream) and self.stream[self.ptr] is not None:
            result.append(self.stream[self.ptr])
            self.ptr += 1
        return result